qa/rpc-tests/buip055.py

```diff
@@ -0,0 +1,490 @@
+        self.sync_all()
+
+    def generateTx(self, node, txBytes, addrs, data=None):
+        wallet = node.listunspent()
+        wallet.sort(key=lambda x: x["amount"], reverse=False)
+        logging.info("Wallet length is %d" % len(wallet))
+
+        size = 0
+        count = 0
+        decContext = decimal.getcontext().prec
+        decimal.getcontext().prec = 8 + 8  # 8 digits to get to 21million, and each bitcoin is 100 million satoshis
+        while size < txBytes:
+            count += 1
+            utxo = wallet.pop()
+            outp = {}
+            # Make the tx bigger by adding addtl outputs so it validates faster
+            payamt = satoshi_round(utxo["amount"] / decimal.Decimal(8.0))
+            for x in range(0, 8):
+                # its test code, I don't care if rounding error is folded into the fee
+                outp[addrs[(count + x) % len(addrs)]] = payamt
+                #outscript = self.wastefulOutput(addrs[(count+x)%len(addrs)])
+                #outscripthex = hexlify(outscript).decode("ascii")
+                #outp[outscripthex] = payamt
+            if data:
+                outp["data"] = data
+            txn = createrawtransaction([utxo], outp, wastefulOutput)
+            # txn2 = node.createrawtransaction([utxo], outp)
+            signedtxn = node.signrawtransaction(txn)
+            size += len(binascii.unhexlify(signedtxn["hex"]))
+            node.sendrawtransaction(signedtxn["hex"])
+        logging.info("%d tx %d length" % (count, size))
+        decimal.getcontext().prec = decContext
+        return (count, size)
+
+    def run_test(self):
+        # Creating UTXOs needed for building tx for large blocks
+        NUM_ADDRS = 50
+        logging.info("Creating addresses...")
+        self.nodes[0].keypoolrefill(NUM_ADDRS)
+        self.nodes[1].keypoolrefill(NUM_ADDRS)
+        addrs = [self.nodes[0].getnewaddress() for _ in range(NUM_ADDRS)]
+        addrs1 = [self.nodes[1].getnewaddress() for _ in range(NUM_ADDRS)]
+        addrs3 = [self.nodes[3].getnewaddress() for _ in range(5)]
+        logging.info("creating utxos")
+
+        self.nodes[1].generate(5)
+        sync_blocks(self.nodes)
+
+        self.createUtxos(self.nodes[0], addrs, 3000)
+        for i in range(0,2):
+            self.createUtxos(self.nodes[1], addrs1, 3000)
+        self.createUtxos(self.nodes[3], addrs3, 3000)
+
+        sync_blocks(self.nodes)
+
+        self.testDefaults()
+        time.sleep(1)  # by controlling the forkTime, we know exactly what the median block will be
+        forkTime = self.testCli()  # also sets up parameters on nodes 0, 1 to to fork
+        time.sleep(2)  # Wait to ensure that every newly generated block is after the fork time.
+
+        base = [x.getblockcount() for x in self.nodes]
+        assert_equal(base, [base[0]] * 4)
+
+        # TEST REQ-3: that a <= 1 MB block is rejected by the fork nodes
+        # the rejection happens for the first block whose nTime is
```

Comment:
what is nTime?  looks like you were about to write it in  "block whose nTime is"  ?